Code/Editor/CryEditDoc.cpp

```diff
@@ -1004,43 +994,45 @@ bool CCryEditDoc::HasLayerNameConflicts()
 bool CCryEditDoc::DoSaveDocument(const QString& filename, TSaveDocContext& context)
 {
     bool& bSaved = context.bSaved;
-    if (bSaved)
+    if (!bSaved)
+    {
+        return bSaved;
+    }
+
+    // Paranoia - we shouldn't get this far into the save routine without a level loaded (empty levelPath)
+    // If nothing is loaded, we don't need to save anything
+    if (filename.isEmpty())
     {
-        // Paranoia - we shouldn't get this far into the save routine without a level loaded (empty levelPath)
-        // If nothing is loaded, we don't need to save anything
-        if (filename.isEmpty())
+        bSaved = false;
+    }
```

Comment:
Wouldn't you want the same early out here, similar de-nesting as you applied above?
It looks like its another early out, where it returns false if the filename is empty.

If you do that then you'll be able to remove the scope and not need an else at all here.

Reply:
It'd have to look something like:
```cpp
if(filename.isEmpty()) 
{
    bSaved = false;
    return false;
}
```
but yes it would reduce the nesting by eliminating the `else` block